testlog_etl/transforms/jscov_to_es.py

```diff
@@ -11,58 +11,79 @@
 
 import json
 
-from pyLibrary.dot import wrap
-from pyLibrary.env.git import get_git_revision
-from testlog_etl.transforms.pulse_block_to_es import scrub_pulse_record
-from testlog_etl.transforms import EtlHeadGenerator
 from pyLibrary.dot import Dict
+from pyLibrary.dot import wrap
+from pyLibrary.debugs.logs import Log
 from pyLibrary.env import http
+from testlog_etl.transforms import EtlHeadGenerator
+from testlog_etl.transforms.pulse_block_to_es import scrub_pulse_record
 
 
 def process(source_key, source, destination, resources, please_stop=None):
-    output = []
+    keys = []
+    records = []
     etl_header_gen = EtlHeadGenerator(source_key)
+    print "Processing " + source_key
+    bucket_file_count = -1
+
+    for msg_line_index, msg_line in enumerate(source.read_lines()):
+        if please_stop:
+            Log.error("Shutdown detected. Stopping job ETL.")
 
-    for i, line in enumerate(source.read_lines()):
         stats = Dict()
-        pulse_record = scrub_pulse_record(source_key, i, line, stats)
+        pulse_record = scrub_pulse_record(source_key, msg_line_index, msg_line, stats)
```

Comment:
Do not use `scrub_pulse_record`:  It was meant to abstract-away many early poor formats.